### ISSUE TITLE
IndexError: bytearray index out of range

### ISSUE THREAD

**Issue by MikeFalcon77** [RandomityGuy/hxDIF#2]
Hi,
It's me again ) 
I've tried to import the last two levels and get:
IndexError: bytearray index out of range
`
    dif = Dif.Load(str(filepath))
  File "C:\Users\Gamebox\AppData\Roaming\Blender Foundation\Blender\2.92\scripts\addons\io_dif\hxDif.py", line 356, in Load
    return Dif.read(br)
  File "C:\Users\Gamebox\AppData\Roaming\Blender Foundation\Blender\2.92\scripts\addons\io_dif\hxDif.py", line 396, in read
    ret.interiors = ReaderExtensions.readArray(io,_hx_local_0)
  File "C:\Users\Gamebox\AppData\Roaming\Blender Foundation\Blender\2.92\scripts\addons\io_dif\hxDif.py", line 1398, in readArray
    x = readMethod(io)
  File "C:\Users\Gamebox\AppData\Roaming\Blender Foundation\Blender\2.92\scripts\addons\io_dif\hxDif.py", line 395, in _hx_local_0
    return Interior.read(io,version)
  File "C:\Users\Gamebox\AppData\Roaming\Blender Foundation\Blender\2.92\scripts\addons\io_dif\hxDif.py", line 1039, in read
    it.hullIndices = ReaderExtensions.readArrayAs(io,_hx_local_36,_hx_local_37,_hx_local_38)
  File "C:\Users\Gamebox\AppData\Roaming\Blender Foundation\Blender\2.92\scripts\addons\io_dif\hxDif.py", line 1421, in readArrayAs
    x1 = failMethod(io)
  File "C:\Users\Gamebox\AppData\Roaming\Blender Foundation\Blender\2.92\scripts\addons\io_dif\hxDif.py", line 1036, in _hx_local_37
    return io.readInt32()
  File "C:\Users\Gamebox\AppData\Roaming\Blender Foundation\Blender\2.92\scripts\addons\io_dif\hxDif.py", line 2492, in readInt32
    v = (((_this.b[pos] | ((_this.b[(pos + 1)] << 8))) | ((_this.b[(pos + 2)] << 16))) | ((_this.b[(pos + 3)] << 24)))
IndexError: bytearray index out of range
`
Blender: 2.92
OS: Windows 10 64-bit
Interior files with issue are attached

[interiors_with_issues.zip](https://github.com/RandomityGuy/hxDIF/files/7470843/interiors_with_issues.zip)

**Comment by RandomityGuy** [comment]
Rough fix here (which will break compatibility with other versions) is to replace all instances of readArrayAs in Interior.read method with `io.readArray(io -> io.read
Int32());` if that doesn't work, undo the change and then replace only the readArrayAs calls that are used to assign values to Interior.hullIndices, hullPlaneIndices, hullEmitString indices to the aforementioned io.readArray function.
I'll do a proper fix that doesn't break the compatibility in a few days

**Comment by MikeFalcon77** [comment]
Hi,

Thank you. Also It seems interior pivot point is wrong, i.e. when I've added exported interior model into the scene and start to add other objects I've discovered that objects positions between each other are correct (comparing to reference level in TGE) but its positions has some shifts relatively to interior object. But I don't know what is a pivot point for original interiors (BSP) in TGE.
